File: precihole/public/py/expense_claim.py
import frappe

# def set_adv_empty(doc,method):
#     doc.set("advances", [])

# def add_adv(doc, method):
#     set_adv_empty(doc, method=None)
#     for i in doc.indent:
#         emp_adv = frappe.db.get_list('Employee Advance',{'indent': i.indent_name},['name', 'posting_date','advance_amount','claimed_amount'])
#         #emp_adv contain list of emp adv for given indent
#         for i in emp_adv:
#             row = doc.append('advances', {})
#             row.employee_advance = i.name
#             row.posting_date = i.posting_date
#             row.advance_paid = i.advance_amount
#             row.unclaimed_amount = i.advance_amount - i.claimed_amount
def update_status_after_submit(doc,method):
    for u in doc.indent:
        if u.indent_name and u.is_partial == 'No':
            receive_status = frappe.db.get_value('Indent', u.indent_name,'receive_status')
            if receive_status == "Fully Received":
                frappe.db.set_value("Indent",u.indent_name, "workflow_state", "Completed")
                frappe.db.set_value("Indent",u.indent_name, "billing_status", "Fully Billed")
        elif u.indent_name and u.is_partial == 'Yes':
                receive_status = frappe.db.get_value('Indent', u.indent_name,'receive_status')
                if receive_status == "Fully Received":
                    frappe.db.set_value("Indent",u.indent_name, "billing_status", "Partially Billed")
def update_status_after_cancel(doc,method):
    for u in doc.indent:
        if u.indent_name and (u.is_partial == 'No' or u.is_partial == 'Yes'):
            expense_claim = frappe.db.get_all('Indent Details',{
                    'indent_name': u.indent_name,
                    'docstatus': 1,
                    'parenttype': "Expense Claim"
                },
                ['parent']
            )
            if expense_claim:
                frappe.db.set_value("Indent",u.indent_name, "workflow_state", "To Bill")
                frappe.db.set_value("Indent",u.indent_name, "billing_status", "Partially Billed")
            elif not expense_claim:
                frappe.db.set_value("Indent",u.indent_name, "workflow_state", "To Bill")
                frappe.db.set_value("Indent",u.indent_name, "billing_status", "Not Billed")

File: precihole/public/js/expense_claim.js
frappe.ui.form.on('Expense Claim',  {
    refresh: function(frm) {
        frm.set_query('employee', function() {
			return {
				'filters': {
					'name': ['=', '']
				}
			};
		});
    }
});
// frappe.ui.form.on('Expense Claim',  {
//     refresh: function(frm) {
//         //frappe.msgprint('child.indent_name')
//         // var site_url;
//         // var child = frm.doc.indent
//         // const emp_adv = []

//         // frappe.run_serially([ 
//         //     () =>   frm.call({
//         //                 //async:false = Code paused. (Other code waiting for this to finish.)
//         //                 //async:true = Code continued. (Nothing gets paused. Other code is not waiting.)
//         //                 async: false,
//         //                 //app path
//         //                 method: "precihole.public.py.purchase_order.get_site_url",
//         //                 callback: function(r) {
//         //                     if(r.message)
//         //                     site_url = r.message.data
                            
//         //                 }
//         //             }),
//         //     () => { 
//         //         for(var t=0;t<child.length;t++){

//         //         }
//         //         //frm.refresh_field('advances')
//         //     },
//         //     () => {
//         //         //var test = ['test','hhh']
//         //         //console.log(emp_adv)
//         //     }
//         // ],
//         // );
//     }
// });

// frappe.ui.form.on('Indent Details', {
// 	indent_name(frm, cdt, cdn) {
//         var child = locals[cdt][cdn];
//         var site_url;
//         var emp_adv = []
//         //frappe.msgprint(child.indent_name)
//         frm.call({
//             //async:false = Code paused. (Other code waiting for this to finish.)
//             //async:true = Code continued. (Nothing gets paused. Other code is not waiting.)
//             async: false,
//             //app path
//             method: "precihole.public.py.purchase_order.get_site_url",
//             callback: function(r) {
//                 if(r.message)
//                 site_url = r.message.data
                
//             }
//         })
//         //frappe.msgprint(site_url)
//         fetch(`${site_url}/api/resource/Employee%20Advance?filters=[["indent",%20"=",%20"${child.indent_name}"]]&fields=["name","posting_date","advance_amount","   "]`, {
//             method: 'GET',
//             headers: {
//                 'Accept': 'application/json',
//                 'Content-Type': 'application/json',
//             }
//         })   
//         .then(r => r.json())s
//         .then(r => {
//             //console.log(r.data)
//             if(r.data !== undefined){
//                 for(var k=0;k<r.data.length;k++){
//                     if (frm.doc.advances == undefined){
//                         cur_frm.add_child("advances", {
//                             employee_advance : r.data[k].name,
//                             posting_date : r.data[k].posting_date,
//                             advance_paid : r.data[k].advance_amount,
//                             unclaimed_amount : r.data[k].advance_amount - r.data[k].claimed_amount
//                         })
//                     }
//                     // else if(frm.doc.advances !== undefined){
//                     //     for(var adv=0;adv<frm.doc.advances.length;adv++){
//                     //         if(frm.doc.advances[adv].employee_advance !== r.data[k].name){
//                     //             cur_frm.add_child("advances", {
//                     //                 employee_advance : r.data[k].name,
//                     //                 posting_date : r.data[k].posting_date,
//                     //                 advance_paid : r.data[k].advance_amount,
//                     //                 unclaimed_amount : r.data[k].advance_amount - r.data[k].claimed_amount
//                     //             })
//                     //         }
//                     //     }
//                     // }
                    
//                 }
//                 frm.refresh_field('advances');
//             }
//         })
// 	}
// })